photonlib-java-examples/aimandrange/src/main/java/frc/robot/Robot.java

```diff
@@ -24,88 +24,139 @@
+    public void robotPeriodic() {
+        // Update drivetrain subsystem
+        drivetrain.periodic();
+
+        // Log values to the dashboard
+        drivetrain.log();
+    }
 
     @Override
-    public void teleopPeriodic() {
-        double forwardSpeed;
-        double rotationSpeed;
+    public void disabledPeriodic() {
+        drivetrain.stop();
+    }
+
+    @Override
+    public void teleopInit() {
+        resetPose();
+    }
 
-        if (xboxController.getAButton()) {
-            // Vision-alignment mode
-            // Query the latest result from PhotonVision
-            var result = camera.getLatestResult();
+    @Override
+    public void teleopPeriodic() {
+        // Calculate drivetrain commands from Joystick values
+        double forward = -controller.getLeftY() * Constants.Swerve.kMaxLinearSpeed;
+        double strafe = -controller.getLeftX() * Constants.Swerve.kMaxLinearSpeed;
+        double turn = -controller.getRightX() * Constants.Swerve.kMaxAngularSpeed;
 
+        // Read in relevant data from the Camera
+        boolean targetVisible = false;
+        double targetYaw = 0.0;
+        double targetRange = 0.0;
+        var results = camera.getAllUnreadResults();
+        if (!results.isEmpty()) {
+            // Camera processed a new frame since last
+            // Get the last one in the list.
+            var result = results.get(results.size() - 1);
             if (result.hasTargets()) {
-                // First calculate range
-                double range =
-                        PhotonUtils.calculateDistanceToTargetMeters(
-                                CAMERA_HEIGHT_METERS,
-                                TARGET_HEIGHT_METERS,
-                                CAMERA_PITCH_RADIANS,
-                                Units.degreesToRadians(result.getBestTarget().getPitch()));
-
-                // Use this range as the measurement we give to the PID controller.
-                // -1.0 required to ensure positive PID controller effort _increases_ range
-                forwardSpeed = -forwardController.calculate(range, GOAL_RANGE_METERS);
-
-                // Also calculate angular power
-                // -1.0 required to ensure positive PID controller effort _increases_ yaw
-                rotationSpeed = -turnController.calculate(result.getBestTarget().getYaw(), 0);
-            } else {
-                // If we have no targets, stay still.
-                forwardSpeed = 0;
-                rotationSpeed = 0;
+                // At least one AprilTag was seen by the camera
+                for (var target : result.getTargets()) {
+                    if (target.getFiducialId() == 7) {
+                        // Found Tag 7, record its information
+                        targetYaw = target.getYaw();
+                        targetRange =
+                                PhotonUtils.calculateDistanceToTargetMeters(
+                                        kRobotToCam.getZ(),
```

Comment:
This line is RLIed to the docs page, but we never really talk about where this kRobotToCam is, where it is coming from, how it is found, etc. in the RLI itself. its sort of intuitive but can be unclear especially given the coordinate systems you have to deal with when getting this value accurately. The concept of the transform may not be necessary to introduce here and may want to be avoided to reduce confusion? could go either way here, its something they will have to know at some point, maybe even a admonition about it would do the trick? Interested to hear your thoughts

Reply:
oooh good point. It's inclusion here is due to:
1) the pose-estimating example had it
2) I copy-pasted the example here
4) I like to think of myself as "good at software" so I don't hardcode things twice in two different spots.

However, in the context of the example, the correct pedagogy is to avoid including it. Will fix.